packages/node/test/integration/secure-deposit.spec.ts

```diff
@@ -39,4 +43,125 @@ describe("Node method follows spec - deposit", () => {
+    const multisigAddress = await createChannel(nodeA, nodeB);
+    const erc20ContractAddress = global["networkContext"]["DolphinCoin"];
+    const erc20Contract = new Contract(
+      erc20ContractAddress,
+      DolphinCoin.abi,
+      new JsonRpcProvider(global["ganacheURL"])
+    );
+
+    let depositReq = makeDepositRequest(
+      multisigAddress,
+      One,
+      erc20ContractAddress
+    );
+
+    expect(
+      await erc20Contract.functions.balanceOf(await nodeA.signerAddress())
+    ).toEqual(Zero);
+
+    let error;
+    try {
+      await nodeA.router.dispatch(depositReq);
+    } catch (e) {
+      error = e;
+    } finally {
+      expect(error).toEqual(
+        INSUFFICIENT_ERC20_FUNDS(await nodeA.signerAddress())
+      );
+    }
+
+    await transferERC20Tokens(await nodeA.signerAddress());
+    await transferERC20Tokens(await nodeB.signerAddress());
+
+    let preDepositBalance = await provider.getBalance(multisigAddress);
+    await nodeA.router.dispatch(depositReq);
+    await nodeB.router.dispatch(depositReq);
+
+    expect(await provider.getBalance(multisigAddress)).toEqual(
+      preDepositBalance
+    );
+    expect(await erc20Contract.functions.balanceOf(multisigAddress)).toEqual(
+      Two
+    );
+
+    await confirmEthAndERC20FreeBalances(
+      nodeA,
+      multisigAddress,
+      erc20ContractAddress
+    );
+    await confirmEthAndERC20FreeBalances(
+      nodeB,
+      multisigAddress,
+      erc20ContractAddress
+    );
+
+    // now deposits ETH
+
+    depositReq = makeDepositRequest(multisigAddress, One);
+
+    preDepositBalance = await provider.getBalance(multisigAddress);
+    await nodeA.router.dispatch(depositReq);
+    await nodeB.router.dispatch(depositReq);
+
+    expect((await provider.getBalance(multisigAddress)).toNumber()).toEqual(
+      preDepositBalance.add(2).toNumber()
+    );
+
+    const freeBalanceState = await getFreeBalanceState(nodeA, multisigAddress);
+    for (const key in freeBalanceState) {
+      expect(freeBalanceState[key]).toEqual(One);
+    }
+  });
 });
+
+/**
+ * @return the ERC20 token balance of the receiver
+ */
+async function transferERC20Tokens(
+  toAddress: string,
+  tokenAddress: string = global["networkContext"]["DolphinCoin"],
+  contractABI: ContractABI = DolphinCoin.abi,
+  amount: BigNumber = One
+): Promise<BigNumber> {
+  const deployerAccount = new Wallet(
+    global["fundedPrivateKey"],
+    new JsonRpcProvider(global["ganacheURL"])
+  );
+
+  const contract = new Contract(tokenAddress, contractABI, deployerAccount);
+
+  const balanceBefore: BigNumber = await contract.functions.balanceOf(
+    toAddress
+  );
+
+  await contract.functions.transfer(toAddress, amount);
+  const balanceAfter: BigNumber = await contract.functions.balanceOf(toAddress);
+
+  expect(balanceAfter.sub(balanceBefore)).toEqual(amount);
+
+  return balanceAfter;
+}
+
+async function confirmEthAndERC20FreeBalances(
```

Comment:
I do not recommend putting expect calls inside helper functions. Better to use helper functions to construct Dayan Andy then place the expect in the test function

Reply:
I don't see what's particularly bad about this. The high-level test sets up the context and is very simple about what it expects to happen, while the logic of the more minute checks is carried out one level down.

If all expects were to be expected at the high level then most test suites can become overly verbose and we lose clarity of how the test achieves what it tries to achieve.

Reply:
We don’t need the same check in every test that uses the helper
Just write a smaller test